src/store/inflight_activation.rs

```diff
@@ -74,10 +81,50 @@ impl From<TaskActivationStatus> for InflightActivationStatus {
     }
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Type)]
+pub enum InflightOnAttemptsExceeded {
+    /// Unused but necessary to align with sentry-protos
+    Unspecified,
+    Discard,
+    Deadletter,
+}
+
+impl From<InflightOnAttemptsExceeded> for OnAttemptsExceeded {
+    fn from(val: InflightOnAttemptsExceeded) -> Self {
+        match val {
+            InflightOnAttemptsExceeded::Unspecified => OnAttemptsExceeded::Unspecified,
+            InflightOnAttemptsExceeded::Discard => OnAttemptsExceeded::Discard,
+            InflightOnAttemptsExceeded::Deadletter => OnAttemptsExceeded::Deadletter,
+        }
+    }
+}
+
+impl TryFrom<i32> for InflightOnAttemptsExceeded {
+    type Error = ();
+
+    fn try_from(v: i32) -> Result<Self, Self::Error> {
+        match v {
+            x if x == InflightOnAttemptsExceeded::Unspecified as i32 => {
+                Ok(InflightOnAttemptsExceeded::Unspecified)
+            }
+            x if x == InflightOnAttemptsExceeded::Discard as i32 => {
+                Ok(InflightOnAttemptsExceeded::Discard)
+            }
+            x if x == InflightOnAttemptsExceeded::Deadletter as i32 => {
+                Ok(InflightOnAttemptsExceeded::Deadletter)
+            }
+            _ => Err(()),
+        }
+    }
+}
+
 #[derive(Clone, Debug, PartialEq)]
 pub struct InflightActivation {
+    /// The unique id of the activation
+    pub id: String,
+
     /// The protobuf activation that was received from kafka
-    pub activation: TaskActivation,
+    pub activation: Option<TaskActivation>,
```

Comment:
I wonder if we should have two structs, one used during ingest to insert, and the other used by the store when reading/updating activation state.